### PR TITLE
Refactor Github Authentication

### PR DESCRIPTION
**Pull request by pat-s**
This PR refactors the way a user authenticates to Github. Rather than using OAuth2.0 and having to eventually cache a token, the user is asked to create a GITHUB_TOKEN/GITHUB_PAT and store this one in their `.Renviron`.

This is a one-time procedure and I do not see any disadvantages currently.

This PR aims to be merge-ready on its own. Therefore, some changes were only made to achieve this goal and are in fact already superseeded by #112 

There is no fixed order in which this and #112 should be merged, either of them requires post-work after the other is merged.

### CODE REVIEW

**Comment by krlmlr** [comment]
Deploy preview for *eloquent-wescoff-5685d5* ready!

Built with commit e4fa6dff396087999a9b7a969d86ec8d191ffad1

https://deploy-preview-113--eloquent-wescoff-5685d5.netlify.com

**Comment by pat-s** [comment]
I had to make a bit more changes to make everything play nicely with each other. 

Among others I removed the `quiet` argument since I do not think that is it of much use. Also it was not used in all functions and brought up some problems when removing certain functions.

**Comment by pat-s** [comment]
@krlmlr any chances to review this in the near future? This blocks to some degree development in _tic_.

**Comment by pat-s** [comment]
@krlmlr 
I've used this branch for some auth calls in the last weeks and think it's stable.

Also I would like to push on the travis.com supp using this PR as the base. 
Do you have capacities to have a look?

Otherwise I would start the travis.com supp branching off from here. However, everything is becoming a bit messy meanwhile with all the open PRs here and in _tic_. Hence I would appreciate if we could get something merged in the next days :)

File: R/travis-vars.R [hunk 1]
```diff
@@ -100,17 +101,15 @@ travis_get_var_id <- function(name, repo = github_repo(),
...
+                           token = auth_travis()) {
   var_id <- travis_get_var_id(
     name = name, repo = repo, token = token
   )
 
   if (!is.null(var_id)) {
-    travis_patch_var(var_id, name, value, public, token, repo, quiet)
-  } else {
-    travis_post_var(name, value, public, token, repo, quiet)
+    travis_delete_var(id = var_id, name = name, token = token, repo = repo)
```
[the remark this answers is not in the document]
Comment (pat-s) [reply R/travis-vars.R]: In my mind this is more clean than "patching". And we have one function less. I don't really see the need for it.

Also patching is not available on any other (CI) system I've interacted with so far.

**Comment by pat-s** [comment]
Superseeded by #115.

**Pull request closed without merging**